pallets/pools/src/lib.rs

```diff
@@ -1524,20 +1625,54 @@ pub mod pallet {
 
 					pool.tranches.combine_with_mut_residual_top(
 						tranches.into_iter(),
-						|tranche, (new_tranche_type, seniority)| {
+						|tranche, tranche_update| {
 							// Update debt of the tranche such that the interest is accrued until now with the previous interest rate
 							tranche.accrue(now)?;
 
-							tranche.tranche_type = new_tranche_type.clone();
+							tranche.tranche_type = tranche_update.tranche_type.clone();
 
-							if let Some(new_seniority) = seniority {
+							if let Some(new_seniority) = tranche_update.seniority {
 								tranche.seniority = new_seniority.clone();
 							}
+
 							Ok(())
 						},
 					)?;
 				}
 
+				//
+				// The case when Metadata AND the tranche changed, we don't allow for an or. Both have to be changed (for now)
+				//
+				if let Change::NewValue(metadata) = &changes.tranche_metadata {
+					for (tranche, updated_metadata) in
+						pool.tranches.tranches.iter().zip(metadata.iter())
+					{
+						let decimals = match T::AssetRegistry::metadata(&tranche.currency) {
+							Some(metadata) => metadata.decimals,
+							None => return Err(Error::<T>::BadMetadata.into()),
+						};
+
+						let parachain_id = T::ParachainId::get();
+						let m = create_asset_metadata(
+							decimals,
+							tranche.currency,
+							parachain_id,
+							updated_metadata.clone().token_name.to_vec(),
+							updated_metadata.clone().token_symbol.to_vec(),
+						)?;
+
+						assert_ok!(T::AssetRegistry::update_asset(
+							tranche.currency,
+							Some(m.decimals),
+							Some(m.name),
+							Some(m.symbol),
```

Comment:
@offerijns are we really sure we want to allow for the decimals, name, and symbol to be updated? This would break cross-chain integrations.

Reply:
Tranche token decimals should just copy the decimals of the pool currency, which in itself could in theory be updated but we probably would just not allow this on a governance level. I still need to take a look at the PR code to review how/whether this is now implemented this way.

Reply:
The decimals is a hard issue, the name and symbol is a soft issue; as in, if those are updated, other chains that may integrate with us and have tranche tokens will may display the outdated name and/or symbol which is a UX problem.

> Regarding the decimals, I think this could really be an issue and would like to make in non-updatable. To easy to miss and easy to add in the future.

💯 agree